app/models/manageiq/providers/ansible_tower/configuration_manager/refresh_parser.rb

```diff
@@ -0,0 +1,25 @@
+module ManageIQ::Providers
+  module AnsibleTower
+    class ConfigurationManager::RefreshParser
+      include Vmdb::Logging
+
+      def self.configuration_inv_to_hashes(inv)
+        new.configuration_inv_to_hashes(inv)
+      end
+
+      def configuration_inv_to_hashes(inv)
+        {:configured_systems => configured_system_inv_to_hashes(inv[:hosts])}
+      end
+
+      def configured_system_inv_to_hashes(recs)
```

Comment:
It's also a LOT less data to juggle :)

As we move forward you may want to consider the `@data / @data_index` pattern from the other providers.